### PR TITLE
feat(button): Add support for icons in CSS buttons

### PR DESCRIPTION
**Pull request by stephanerangaya**
Closes https://github.com/Workday/canvas-kit/issues/350

This PR adds support for icons in CSS Button to match the React equivalents. It relies on the existing CSS Icon module.

### CODE REVIEW

File: modules/button/css/README.md [hunk 1]
```diff
@@ -291,6 +292,31 @@ Apply primary styling to the button by adding `.wdc-btn-primary` to the buttons
 <button class="wdc-btn-text wdc-btn-text-inverse">Hovered Button</button>
 ```
 
+### Inline Icon
+You can use a [System Icon](/?path=/story/css-icon--system-icon) either before or after the text of Primary, Secondary, Delete and Text buttons. The class `.wdc-icon-position-left` (default) or  `.wdc-icon-position-right` adjust the margin of the icon.
```
Comment (anicholls) [inline modules/button/css/README.md]: Can we refer to the readme instead?

```suggestion
You can use a [System Icon](../../icon/css/README.md) either before or after the text of Primary, Secondary, Delete and Text buttons. The class `.wdc-icon-position-left` (default) or  `.wdc-icon-position-right` adjust the margin of the icon.
```

File: modules/button/css/README.md [hunk 2]
```diff
@@ -291,6 +292,31 @@ Apply primary styling to the button by adding `.wdc-btn-primary` to the buttons
...
+```
+
+```html
+<button className="wdc-btn wdc-btn-primary wdc-btn-size-l">
+  <i className="wdc-icon" data-icon="edit" data-category="system" />Edit
+</button>
+
+
+<a href="#" className="wdc-btn-text">
+  <i className="wdc-icon wdc-icon-position-left" data-icon="edit" data-category="system" /> Edit
```
Comment (anicholls) [inline modules/button/css/README.md]: Can we drop this example? It's essentially the same as the first. I think we only need the first and third

File: modules/button/css/lib/button.scss [hunk 3]
```diff
@@ -49,32 +60,58 @@ $_wdc-btn-caret-down-icon-white: svg-load(
   &.wdc-btn-hover,
   &:hover {
     background-color: $wdc-btn-bg-secondary-hover;
+    @include wdc-icon-color(#333d47);
```
Comment (anicholls) [inline modules/button/css/lib/button.scss]: These should all be Canvas colors

File: modules/button/css/lib/button.scss [hunk 4]
```diff
@@ -49,32 +60,58 @@ $_wdc-btn-caret-down-icon-white: svg-load(
...
+    @include wdc-icon-color($wdc-btn-secondary-disabled-text-color);
   }
 
   &.wdc-btn-grow {
     width: 100%;
     max-width: 100%;
   }
 
+  .wdc-icon[data-category="system"]:not([data-circle-background]) {
+    vertical-align: -7px;
```
Comment (anicholls) [inline modules/button/css/lib/button.scss]: What does this do? It doesn't seem to make a difference if I remove it.

File: modules/button/css/lib/button.scss [hunk 5]
```diff
@@ -49,32 +60,58 @@ $_wdc-btn-caret-down-icon-white: svg-load(
...
+
+    &,
+    .wdc-icon-position-left & {
+      margin: 0 $wdc-spacing-xxs;
+      margin-left: 0;
+    }
+  
+    .wdc-icon-position-right & {
+      margin: 0 $wdc-spacing-xxs;
```
Comment (anicholls) [inline modules/button/css/lib/button.scss]: Nit: Can we either use `margin-right: $wdc-spacing-xxs;` or `margin: 0 0 0 $wdc-spacing-xxs;`? Same goes elsewhere.

File: modules/button/css/stories.tsx [hunk 6]
```diff
@@ -1,10 +1,15 @@
...
+import {Component} from 'react';
 import {storiesOf} from '@storybook/react';
 import withReadme from 'storybook-readme/with-readme';
 import {SystemIcon} from '@workday/canvas-kit-react-icon';
-import {activityStreamIcon} from '@workday/canvas-system-icons-web';
+import {
+  activityStreamIcon,
+} from '@workday/canvas-system-icons-web';
 import README from './README.md';
+// import initializeIcons from '../../icon/css/lib/canvas-kit-css-icon';
```
Comment (anicholls) [inline modules/button/css/stories.tsx]:
```suggestion
```

File: modules/button/css/lib/button.scss (hunk 4 above)
Comment (stephanerangaya) [reply modules/button/css/lib/button.scss]: This is for the medium size

Comment (anicholls) [reply modules/button/css/lib/button.scss]: Can you describe what it's actually doing though? Why is it needed?

File: modules/button/css/lib/button.scss [hunk 7]
```diff
@@ -124,59 +173,71 @@ $_wdc-btn-caret-down-icon-white: svg-load(
...
     }
 
     &.wdc-btn-disabled,
     &:disabled {
       box-shadow: none;
       background-color: $wdc-btn-bg-primary-disabled;
       color: $wdc-btn-text-color;
+      @include wdc-icon-color($wdc-btn-text-color);
     }
+
```
Comment (anicholls) [inline modules/button/css/lib/button.scss]:
```suggestion
```

File: modules/button/css/lib/button.scss [hunk 8]
```diff
@@ -112,6 +147,20 @@ $_wdc-btn-caret-down-icon-white: svg-load(
     height: 48px;
     padding: 0 $wdc-spacing-l;
     font-size: 16px;
+    .wdc-icon[data-category="system"]:not([data-circle-background]) {
+      vertical-align: -6px;
+      
+      &,
+      &.wdc-icon-position-left {
+        margin: 0 $wdc-spacing-xs;
```
Comment (anicholls) [inline modules/button/css/lib/button.scss]:
```suggestion
        margin: 0 $wdc-spacing-xs 0 -$wdc-spacing-xxxs;
```

File: modules/button/css/lib/button.scss [hunk 9]
```diff
@@ -112,6 +147,20 @@ $_wdc-btn-caret-down-icon-white: svg-load(
...
+        margin-left: -$wdc-spacing-xxxs;
```
Comment (anicholls) [inline modules/button/css/lib/button.scss]:
```suggestion
```

File: modules/button/css/lib/button.scss [hunk 10]
```diff
@@ -112,6 +147,20 @@ $_wdc-btn-caret-down-icon-white: svg-load(
...
+      }
+    
+      &.wdc-icon-position-right {
+        margin: 0 $wdc-spacing-xs;
```
Comment (anicholls) [inline modules/button/css/lib/button.scss]:
```suggestion
        margin: 0 -$wdc-spacing-xxxs 0 $wdc-spacing-xs;
```

File: modules/button/css/lib/button.scss [hunk 11]
```diff
@@ -112,6 +147,20 @@ $_wdc-btn-caret-down-icon-white: svg-load(
...
+        margin-right: -$wdc-spacing-xxxs;
```
Comment (anicholls) [inline modules/button/css/lib/button.scss]:
```suggestion
```

File: modules/button/css/lib/text-button.scss [hunk 12]
```diff
@@ -11,14 +11,31 @@
   border-radius: $wdc-border-radius-m;
   display: inline-flex;
   align-items: center;
+  @include wdc-icon-color($wdc-btn-text);
+
+  .wdc-icon[data-category="system"]:not([data-circle-background]) {
+    &,
+    &.wdc-icon-position-left {
+      margin: 0 $wdc-spacing-xxs;
```
Comment (anicholls) [inline modules/button/css/lib/text-button.scss]:
```suggestion
      margin: 0 $wdc-spacing-xxs 0 0;
```

File: modules/button/css/lib/text-button.scss [hunk 13]
```diff
@@ -11,14 +11,31 @@
...
+      margin-left: 0;
```
Comment (anicholls) [inline modules/button/css/lib/text-button.scss]:
```suggestion
```

File: modules/button/css/lib/text-button.scss [hunk 14]
```diff
@@ -11,14 +11,31 @@
...
+    }
+  
+    &.wdc-icon-position-right {
+      margin: 0 $wdc-spacing-xxs;
```
Comment (anicholls) [inline modules/button/css/lib/text-button.scss]:
```suggestion
      margin: 0 0 0 $wdc-spacing-xxs;
```

File: modules/button/css/lib/text-button.scss [hunk 15]
```diff
@@ -11,14 +11,31 @@
...
+      margin-right: 0;
```
Comment (anicholls) [inline modules/button/css/lib/text-button.scss]:
```suggestion
```

File: modules/button/css/lib/button.scss (hunk 4 above)
Comment (stephanerangaya) [reply modules/button/css/lib/button.scss]: Here's how the medium primary button with icon looks like without this: 
image

Comment (anicholls) [reply modules/button/css/lib/button.scss]: Cool, thank you! Just wanted to make sure it was necessary as I didn't notice this when I disabled it 👍

**Pull request merged**